src/openforms/contrib/reference_lists/client.py

```diff
@@ -0,0 +1,125 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from functools import partial
+from typing import Optional
+
+from django.core.cache import cache
+from django.utils import timezone
+from django.utils.functional import cached_property
+from django.utils.translation import get_language
+
+from ape_pie import APIClient
+from zgw_consumers.service import pagination_helper
+
+from .typing import APITable, APITableItem
+
+REFERENCE_LISTS_LOOKUP_CACHE_TIMEOUT = 5 * 60
+
+
+@dataclass
+class EndDateMixin:
+    @cached_property
+    def parsed_end_date(self) -> datetime | None:
+        expires_on = getattr(self, "expires_on", None)
+        if expires_on is None:
+            return None
+        return expires_on
+
+    @cached_property
+    def is_expired(self) -> bool:
+        if self.parsed_end_date:
+            return self.parsed_end_date <= timezone.now()
+        return False
+
+    @cached_property
+    def is_nearly_expired(self) -> bool:
+        if self.parsed_end_date:
+            return self.parsed_end_date <= (timezone.now() + timedelta(days=7))
+        return False
+
+
+@dataclass
+class TableItem(EndDateMixin):
+    code: str
+    name: str
+    expires_on: Optional[datetime] = None  # ISO 8601 datetime string
```

Comment:
```suggestion
    expires_on: datetime | None = None  # ISO 8601 datetime string
```
please use the modern syntax :) it's more explicit/obvious what it means